### PR TITLE
Do not always mention installation (bnc#941403, bnc#941739)

### PR DESCRIPTION
**Pull request by ancorgs**
I have not even run this code, but since I'm having problems to get my SP1 updated/working...

### CODE REVIEW

File: src/lib/registration/ui/abort_confirmation.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+      # the constructor
+      def initialize
+        textdomain "registration"
+      end
+
+      # displays the popup
+      # @return [Boolean] true whether the user confirmed he/she wants to abort
+      def run
+        if Yast::Mode.installation || Yast::Mode.repair
```
Comment (lslezak) [inline src/lib/registration/ui/abort_confirmation.rb]: The `repair` mode is obsolete, it's not used anymore (the repair module has been dropped).

File: src/lib/registration/ui/abort_confirmation.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          Yast::Popup.ConfirmAbort(:painless)
+        else
+          msg = _("Really abort the registration?\n\n" \
```
Comment (lslezak) [inline src/lib/registration/ui/abort_confirmation.rb]: I'd probably prefer simple "Rally abort?" question with `[Yes][No]` buttons.

- The `no changes will be done to the system` statement is not completely true, registration workflow has several steps, e.g. you register the base product, then can optionally register and install an extension. But when you abort the extension selection dialog, the system has been already registered (=changed)

- `Really abort the *registration*?` is also not completely correct, some registration parts are called from the online migration module, distinguishing regular registration or online migration would make it complex.

Therefore just the simple "Rally abort?" is the best IMHO...

Comment (ancorgs) [inline src/lib/registration/ui/abort_confirmation.rb]: Ok. I was completely unsure about all that, so I decided to "clone" the installation abort message and let you spot the inconsistencies. I'll simplify the text.

File: src/lib/registration/ui/abort_confirmation.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      end
+
+      # displays the popup
+      # @return [Boolean] true whether the user confirmed he/she wants to abort
+      def run
+        if Yast::Mode.installation
+          Yast::Popup.ConfirmAbort(:painless)
+        else
+          # Use .AnyQuestion instead of .YesNo or .ReallyAbort to have full control
+          # on :focus_no and be consistent with the .ConfirmAbort behavior
```
Comment (lslezak) [inline src/lib/registration/ui/abort_confirmation.rb]: Off-topic: Ideally we should add a new generic method for popups and take advantage of optional parameters in Ruby, something like `Popup.show(message: "msg", default_button: :no, header: "label",...)`...

**Comment by lslezak** [comment]
Could you add a (simple) test case?

File: src/lib/registration/ui/abort_confirmation.rb (hunk 3 above)
Comment (ancorgs) [inline src/lib/registration/ui/abort_confirmation.rb]: Absolutely. We have a lot of methods in `Popup` that are different on each other just by subtle and silly small differences. The API looks completely weird and it took me a while to figure out the best way to imitate `ConfirmAbort`.

**Comment by ancorgs** [comment]
Good point. It should be fairly easy to add one.

**Comment by lslezak** [comment]
LGTM, thanks a lot! :+1:

**Pull request merged**